Tribler/Test/Core/CreditMining/test_creditmining.py

```diff
@@ -0,0 +1,252 @@
+        self.choked = True
+        self.total_upload = 0
+        self.total_download = 0
+        self.progress = 0
+        self.pieces = 0
+        self.remote_dl_rate = 0
+        self.country = "ID"
+        self.connection_type = 0
+        self.seed = 1
+        self.upload_only = 1
+        self.read_state = False
+        self.write_state = False
+
+    def setvalue(self, upload_only, uinterested, completed):
+        self.upload_only = upload_only
+        self.remote_interested = uinterested
+        self.progress = 1 if completed else 0
+
+
+class TestBoostingManagerUtilities(TestAsServer):
+    """
+    Test several utilities used in credit mining
+    """
+
+    def __init__(self, *argv, **kwargs):
+        super(TestBoostingManagerUtilities, self).__init__(*argv, **kwargs)
+
+        self.peer = [None] * 6
+        self.peer[0] = DummyPeer(1, "ip1")
+        self.peer[0].setvalue(True, True, True)
+        self.peer[1] = DummyPeer(2, "ip2")
+        self.peer[1].setvalue(False, False, True)
+        self.peer[2] = DummyPeer(3, "ip3")
+        self.peer[2].setvalue(True, False, True)
+        self.peer[3] = DummyPeer(4, "ip4")
+        self.peer[3].setvalue(False, True, False)
+        self.peer[4] = DummyPeer(5, "ip5")
+        self.peer[4].setvalue(False, True, True)
+        self.peer[5] = DummyPeer(6, "ip6")
+        self.peer[5].setvalue(False, False, False)
+
+    def setUp(self, autoload_discovery=True):
+        super(TestBoostingManagerUtilities, self).setUp()
+        self.bsettings = bm.BoostingSettings(self.session)
+        self.bsettings.credit_mining_path = self.session_base_dir
+        self.bsettings.config_file = 'None'
+
+
+    def setUpPreSession(self):
+        super(TestBoostingManagerUtilities, self).setUpPreSession()
+        self.config.set_libtorrent(True)
+        self.config.set_mainline_dht(True)
+
+    def test_translate_peer_info(self):
+        """
+        test - predict number of seeder and leecher only based on peer discovered and
+        their activities
+        """
+        peerlist_dict = []
+        for peer in self.peer:
+            peerlist_dict.append(LibtorrentDownloadImpl.create_peerlist_data(peer))
+
+        num_seed, num_leech = utilities.translate_peers_into_health(peerlist_dict)
+        self.assertEqual(num_seed, 4, "Seeder number don't match")
+        self.assertEqual(num_leech, 3, "Leecher number don't match")
+
+    def test_levenshtein(self):
+        """
+        test levenshtein between two string (in this case, file name)
+
+        source :
+        http://people.cs.pitt.edu/~kirk/cs1501/Pruhs/Fall2006/Assignments/editdistance/Levenshtein%20Distance.htm
+        """
+        string1 = "GUMBO"
+        string2 = "GAMBOL"
+
+        dist = bm.lev(string1, string2)
+        dist_swap = bm.lev(string2, string1)
+
+        self.assertEqual(dist, 2, "Wrong levenshtein distance")
+        self.assertEqual(dist_swap, 2, "Wrong levenshtein distance")
```

Comment:
I guess some more tests can be added? I.e. the same string, totally different strings etc